Add a recipe for liblouis mode

egli opened the pull request:
### Brief summary of what the package does

Liblouis mode is an Emacs major mode for editing [liblouis](https://github.com/liblouis/liblouis) braille translation tables.

### Direct link to the package repository

https://github.com/liblouis/liblouis-mode

### Your association with the package

I am the maintainer of the package

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
  - there is one warning that I don't know how to avoid: I use `eval-after-load` to set the `compilation-error-regexp-alist`
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this!  Quick first pass --

`liblouis.el` with byte-compile using Emacs 27.1:
```
liblouis.el:43:18:Warning: reference to free variable
    `compilation-error-regexp-alist'
liblouis.el:43:18:Warning: assignment to free variable
    `compilation-error-regexp-alist'
```
These are easy to fix with `(require 'compile)` (and then your `package-lint` issue below can disappear)

`liblouis.el` with package-lint 20220213.1457:
```
1 issue found:
42:1: warning: `eval-after-load' is for use in configurations, and should rarely be used in packages.
```
One of the reasons I believe package-lint complains about this one is to discourage packages with user-invisible side effects at load time.  Occasionally there are exceptions, like conventions around adding to `auto-mode-alist`.

egli commented:
Hi @riscy thanks for the review.

I updated `liblouis.el` and fixed the issues you mentioned. I also added a github action to run `package-lint` and `checkdoc` on the elisp, see https://github.com/liblouis/liblouis-mode/actions/workflows/elisp-lint.yml

riscy commented:
Thanks!  This looks great.  As I take another look one (very minor) suggestion would be to make the dep on `'regex-opt` explicit, since there's no guarantee it will always be transitively brought in via `(require 'compile)`.  Extremely unlikely to cause issues, but I thought I'd mention it.
